tests/unit/s2n_self_talk_nonblocking_test.c

```diff
@@ -58,19 +67,23 @@ int mock_client(struct s2n_test_io_pair *io_pair, uint8_t *expected_data, uint32
 
     /* Receive 10MB of data */
     uint32_t remaining = size;
-    while(remaining) {
+    while(remaining && (start_time - time(0)) < max_wait_time) {
         int r = s2n_recv(client_conn, ptr, remaining, &blocked);
         if (r < 0) {
             return 1;
         }
         remaining -= r;
         ptr += r;
+        if ((size - remaining) > (size * send_at_least_percent) && should_block) {
+           raise(SIGSTOP);
```

Comment:
This "did we at least send _some_ data?" condition is also repeated a couple times, so is probably worth a helper function or macro for readability.

Reply:
I have just the thing, I think.